File: src/main/java/com/myimages/service/EventService.java
package com.myimages.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myimages.domain.Event;
import com.myimages.dto.EventDTO;
import com.myimages.dto.EventSearchDTO;
import com.myimages.dto.EventPageDTO;
import com.myimages.dto.EventConvertCriteriaDTO;
import com.myimages.service.GenericService;
import com.myimages.dto.common.RequestDTO;
import com.myimages.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface EventService extends GenericService<Event, Integer> {

	List<Event> findAll();

	ResultDTO addEvent(EventDTO eventDTO, RequestDTO requestDTO);

	ResultDTO updateEvent(EventDTO eventDTO, RequestDTO requestDTO);

    Page<Event> getAllEvents(Pageable pageable);

    Page<Event> getAllEvents(Specification<Event> spec, Pageable pageable);

	ResponseEntity<EventPageDTO> getEvents(EventSearchDTO eventSearchDTO);
	
	List<EventDTO> convertEventsToEventDTOs(List<Event> events, EventConvertCriteriaDTO convertCriteria);

	EventDTO getEventDTOById(Integer eventId);


}

File: myimages_ui/src/services/EventService.js
import http from "../http-common"; 

class EventService {
  getAllEvents(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/event/events`, searchDTO);
  }
 

  get(eventId) {
    return this.getRequest(`/event/${eventId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/event?field=${matchData}`, null);
  }

  addEvent(data) {
    return http.post("/event/addEvent", data);
  }

  update(data) {
  	return http.post("/event/updateEvent", data);
  }
  
  uploadImage(data,eventId) {
  	return http.postForm("/event/uploadImage/"+eventId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new EventService();
